### PR TITLE
Move OCM SDK handler methods to pkg/ocm.go

### PR DESCRIPTION
**Pull request by hectorakemp**
### What type of PR is this?
_(cleanup)_

### What this PR does / why we need it?
- Reorganizes the `upgrade_policies.go` handler. Instead of declaring OCM SDK methods within the handler itself, it moves them to the `pkg/ocm.go` parent package. This allows them to be called by other handler methods cleanly that aren't the upgrade policies handler. It also makes the code a bit nicer to deal with. 
### Which Jira/Github issue(s) this PR fixes?

_Fixes #_
https://issues.redhat.com/browse/OSD-20778

### Pre-checks (if applicable):
- [ ] Tested latest changes against a cluster
- [x] Ran `make generate` command locally to validate code changes
- [ ] Included documentation changes with PR

### CODE REVIEW

File: pkg/ocm/ocm.go [hunk 1]
```diff
@@ -130,6 +137,82 @@ func NewOcmClient(ocmConnection *sdk.Connection) OCMClient {
...
 
+// https://pkg.go.dev/github.com/openshift-online/ocm-sdk-go@v0.1.382/clustersmgmt/v1#Cluster
+func (o *ocmClientImpl) GetCluster(clusterID string) (*cmv1.Cluster, string, error) {
+	log.Debugf("Sending get cluster object request to OCM API: %s", clusterID)
+	request := o.ocmConnection.ClustersMgmt().V1().Clusters().Cluster(clusterID)
+	resp, err := request.Get().Send()
+	if err != nil {
+		return nil, resp.Header().Get(OcmOperationIdHeader), err
+	}
+	return resp.Body(), resp.Header().Get(OcmOperationIdHeader), nil
```
Comment (typeid) [inline pkg/ocm/ocm.go:148]: **suggestion [repeated for other functions containing ocm requests]:** We should probably check `resp.Status()` here: we could in theory have a HTTP request that didn't result in an error (the request itself worked), but the status could be a 500. 

The other functions here do:
```
	// Check the response status code
	if response.Status() < 200 && response.Status() >= 300 {
		return fmt.Errorf("unexpected status: %d", response.Status())
	}
```

**Review by typeid — COMMENTED** [review]
Very nice cleanup, it's looking great @hectorakemp! I added one small suggestion, but other than that it looks good to go.

File: pkg/ocm/ocm.go (hunk 1 above)
Comment (hectorakemp) [reply pkg/ocm/ocm.go:152]: @typeid - this is already handled by the OCM SDK (and returns an error). This holds for all of the HTTP transports methods in the OCM SDK. See here: https://github.com/search?q=repo%3Aopenshift-online%2Focm-sdk-go%20errors.UnmarshalErrorStatus&type=code

Comment (hectorakemp) [reply pkg/ocm/ocm.go:152]: I think that this also means that you can remove these lines from `SendLimitedSupport`, `RemoveLimitedSupport`, etc.

Comment (typeid) [reply pkg/ocm/ocm.go:148]: If I read it correctly, this only handles `result.status >= 400`, what about other status codes?

Comment (hectorakemp) [reply pkg/ocm/ocm.go:154]: Ah, you are indeed correct @typeid. Great pickup :) 

I've added 3xx error handling in `pkg/ocm.go`, some basic tests, as well fixed a type in the conditionals for the service log/limited support functions. 
```
		if resp.Status() < 200 && resp.Status() >= 300 {
```
was never being reached for 3xx responses. I've updated it to 
```
		if resp.Status() < 200 || resp.Status() >= 300 {
```
for all methods in `pkg/ocm.go`.

**Pull request merged**